deploy-deps.sh

```diff
@@ -27,6 +27,16 @@ check_req(){
         fi
     done
 
+    # check if kubectl has server-side support
+    if command -v kubectl &> /dev/null; then
```

Comment:
What error message will you get if you don't have kubectl installed?
I think it will make sense to add this block below the one that check if the command exists.

Reply:
If you get to the `kubectl apply --help | grep -q -- --server-side` line and kubectl is missing, will the script exit immediately or get to print the proper error message?